app/services/api/public/v1/course_search_service.rb

```diff
@@ -0,0 +1,79 @@
+module API
```

Comment:
Why do we need a separate search service for public/v1? May need to have a chat to align a bit on what we're doing.